packages/slate/src/transforms-node/merge-nodes.ts

```diff
@@ -133,12 +133,7 @@ export const mergeNodes: NodeTransforms['mergeNodes'] = (
     // prevent losing formatting when deleting entire nodes when you have a
     // hanging selection.
     // if prevNode is first child in parent,don't remove it.
-    if (
-      (Element.isElement(prevNode) && Editor.isEmpty(editor, prevNode)) ||
-      (Text.isText(prevNode) &&
-        prevNode.text === '' &&
-        prevPath[prevPath.length - 1] !== 0)
-    ) {
+    if (editor.shouldMergeNodesRemovePrevNode(prev, current)) {
```

Comment:
Since we've got this on the Editor interface, should we use that here?

```suggestion
    if (Editor.shouldMergeNodesRemovePrevNode(editor, prev, current)) {
```

Reply:
You're right ,This is my first PR on here. Thank for your guidance